ground/src/main/java/com/google/android/ground/system/LocationManager.kt

```diff
@@ -15,26 +15,24 @@
  */
 package com.google.android.ground.system
 
-import android.Manifest.permission
 import android.location.Location
 import com.google.android.gms.location.LocationRequest
 import com.google.android.ground.coroutines.ApplicationScope
-import com.google.android.ground.repository.MapStateRepository
+import com.google.android.ground.system.channel.ChannelLocationCallback
 import com.google.android.ground.system.rx.RxFusedLocationProviderClient
-import com.google.android.ground.system.stateflow.StateFlowLocationCallback
 import io.reactivex.Single
 import javax.inject.Inject
 import javax.inject.Singleton
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.flow.*
-import kotlinx.coroutines.rx2.await
 import timber.log.Timber
 
 private const val UPDATE_INTERVAL: Long = 1000 /* 1 sec */
 
 private const val FASTEST_INTERVAL: Long = 250 /* 250 ms */
 
-private val FINE_LOCATION_UPDATES_REQUEST =
+val FINE_LOCATION_UPDATES_REQUEST: LocationRequest =
```

Comment:
Can type be inferred here?

Reply:
I got this warning when the type was inferred previously "Declaration has type inferred from a platform call, which can lead to unchecked nullability issues"

Reply:
ah, ok!